src/compositor/lipstickcompositorwindow.cpp

```diff
@@ -68,6 +70,42 @@ LipstickCompositorWindow::~LipstickCompositorWindow()
     LipstickCompositor::instance()->windowDestroyed(this);
 }
 
+void LipstickCompositorWindow::updatePolicyApplicationId()
+{
+    if (m_processId <= 0) {
+        return;
+    }
+
+    QString statFile = QString::fromLatin1("/proc/%1/stat").arg(m_processId);
+    QFile f(statFile);
+    if (!f.open(QIODevice::ReadOnly)) {
+        qWarning() << Q_FUNC_INFO << "Cannot open proc stat for" << statFile;
+        return;
+    }
+
+    // stat line values are split by ' ' in /proc/*/stat
+    QByteArray data = f.readAll();
+    QList<QByteArray> statFields = data.split(' ');
```

Comment:
Late side remark: While it is unlikely, 'comm' field can have spaces and whatnot a la  "/path/to/weird )( name" -> "12345 (weird )( name) R ..." -> looking for last occurrence of ")" or so and doing whitespace splitting from there onward would avoid possible off by N indexing issues below.